### ISSUE TITLE
FVD calculation

### ISSUE THREAD

**Issue by oszilevi** [JunyaoHu/common_metrics_on_video_quality#1]
hi, 
how come you calculate the fvd score with the output of the logits layer? Doesnt the idea of the FID and FVD come from using the feature space?

**Comment by JunyaoHu** [comment]
Thank you for your question.

# TLDR

there is no need to use the function `get_fvd_logits`.

# our implementation
In our calculation process `calculate_fvd.py`,

- **Step 1**: preprocess the video to ensure the proper shape and value range. 

https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/241f953dda359eb3d808f1425b934a9f133032f5/calculate_fvd.py#L30-L31

- **Step 2**: use pre-trained model (i3d) to get the feature space tensor of the video.

https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/241f953dda359eb3d808f1425b934a9f133032f5/calculate_fvd.py#L47-L48

- **Step 3**: calculate FVD score

https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/241f953dda359eb3d808f1425b934a9f133032f5/calculate_fvd.py#L51

# about Step 2

In our function `get_fvd_feats`, it calls the function `get_feats`.

https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/241f953dda359eb3d808f1425b934a9f133032f5/fvd/fvd.py#L53-L57

the input of the function `get_feats` is the video, and the output is the i3d embedding. and the parameter `return_features=True` makes the output as features not logits.

https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/241f953dda359eb3d808f1425b934a9f133032f5/fvd/fvd.py#L43-L50

there is no need to use the function `get_fvd_logits`. I didn't comment out this function, and I'm sorry if that misinterpreted you, but this code is redundant.

**Comment by JunyaoHu** [comment]
Hello, @oszilevi, does my answer explain this question clearly? You can use my recently updated repository, which has been stripped of redundant code. 😊

**Comment by oszilevi** [comment]
yes! Thank u very much 👍🏻

**Comment by JunyaoHu** [comment]
Hello, I updated the repo just now, and it can support 2 pytorch FVD calculation methods (styleganv and videogpt). 

As you say, the method calculate _'the fvd score with the output of the logits layer'_ is the implementation of [videogpt](https://github.com/wilson1yan/VideoGPT/blob/master/videogpt/fvd/fvd.py). 

I must say videogpt is not wrong, instead, is also right, because its function name is not good.

Actually, **logits are features**

in origin I3D model, the tail of the model structure is  'Mixed_5c', 'Logits', and 'Predictions'. [here](https://github.com/google-deepmind/kinetics-i3d/blob/master/i3d.py)

![image](https://github.com/JunyaoHu/common_metrics_on_video_quality/assets/67564714/75265aae-0217-4e7e-b623-71329d1a8af5)

The two model both remove the module of 'Predictions' (softmax):

- As for i3d_pretrained_400.pt, it use the [I3D model file](https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/main/fvd1/videogpt/pytorch_i3d.py) in our repo, the model end with 'Logits', without 'Predictions' (softmax). So it is right.
- As for i3d_torchscript.pt, it use the parameter `return_features=True`. So is it right.

---

Finally, in our repo, I copied the file with the original function name, only importing it with an alias to refer to it.

https://github.com/JunyaoHu/common_metrics_on_video_quality/blob/70984717851a3141924c15a2a93ff50f3e4fa734/calculate_fvd.py#L15-L22

---

I apologize for my incomplete and incorrect answers.
